main.cpp

```diff
@@ -32,28 +32,517 @@ Purpose:  This project continues developing Project3.
  
  */
 
+#include <iostream>
+
 /*
  copied UDT 1:
  */
 
+ //--------------------------------------------------------
+// DishWasher
+//--------------------------------------------------------
+
+struct DishWashingProcess
+{
+    int errorAtMinute;
+    DishWashingProcess(int m) : errorAtMinute(m) {} 
+};
+
+struct DishWasher
+{
+    float washingRoomVolume {5.2f};
+    int numOfPrograms {12};
+    int maxWashingDuration {120};
+    float maxWashTemperature {90.0f};
+    int timerDuration {360};
+    int waitMilliSec {0};
+
+    DishWasher();
+    ~DishWasher();
+
+    bool washDishes(int programNumber);
+    bool dryDishes(int duration, float temperature);
+    void startAtTimer(int startMinutesLater); 
+    float printMaxDishElements(); 
+    DishWashingProcess checkErrors(int errorTime);
+
+};
+
+
+DishWasher::DishWasher() {}
+
```

Comment:
`make the destructors do something like print out the name of the class`
- [ ] revise